libraries/AP_MSP/AP_MSP.cpp

```diff
@@ -102,7 +102,7 @@ void AP_MSP::init()
     for (uint8_t protocol_instance=0; protocol_instance<MSP_MAX_INSTANCES-_msp_status.backend_count; protocol_instance++) {
```

Comment:
insn't the issue just : 
`for (uint8_t protocol_instance=_msp_status.backend_count; protocol_instance<MSP_MAX_INSTANCES; protocol_instance++)`

Reply:
mhmm I don't think so, shouldn't we check each serial protocol for all instances starting at 0 up to the MAX allowed?